server/mscalendar/availability.go

```diff
@@ -441,12 +447,7 @@ func (m *mscalendar) setStatusOrAskUser(user *store.User, currentStatus *model.S
 }
 
 func (m *mscalendar) GetCalendarEvents(user *User, start, end time.Time) (*remote.ViewCalendarResponse, error) {
-	err := m.Filter(withActingUser(user.MattermostUserID))
-	if err != nil {
-		return nil, errors.Wrap(err, "error withRemoteUser")
-	}
-
-	err = m.Filter(withClient)
+	err := m.Filter(withClient)
```

Comment:
Do you see an opportunity to make this code less error-prone? It concerns me that it's easy to make an error based on where this call is placed

Reply:
That was the intention when I removed the `withActingUser` from this method. Now we can call this one without problems from other places since the new `withActingUser` only gets called from the fenced code in the "main" calls.

Reply:
Right, but what can we do to avoid needing to have `withActingUser` in the fenced code? Just want to make sure we don't run into something similar later, especially if it's something not caught by a unit test